src/Utilities/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysis.cs

```diff
@@ -0,0 +1,77 @@
+    using PropertySetAnalysisDomain = MapAbstractDomain<AbstractLocation, PropertySetAbstractValue>;
+    using PointsToAnalysisResult = DataFlowAnalysisResult<PointsToBlockAnalysisResult, PointsToAbstractValue>;
+
+    /// <summary>
+    /// Dataflow analysis to track <see cref="PropertySetAbstractValue"/> of <see cref="AbstractLocation"/>/<see cref="IOperation"/> instances.
+    /// </summary>
+    internal partial class PropertySetAnalysis : ForwardDataFlowAnalysis<PropertySetAnalysisData, PropertySetAnalysisContext, PropertySetAnalysisResult, PropertySetBlockAnalysisResult, PropertySetAbstractValue>
+    {
+        public static readonly PropertySetAnalysisDomain BinaryFormatterAnalysisDomainInstance = new PropertySetAnalysisDomain(PropertySetAbstractValueDomain.Default);
+
+        private PropertySetAnalysis(PropertySetAnalysisDomain analysisDomain, PropertySetDataFlowOperationVisitor operationVisitor)
+            : base(analysisDomain, operationVisitor)
+        {
+        }
+
+        public static ImmutableDictionary<OperationMethodKey, PropertySetAbstractValue> GetOrComputeHazardousParameterUsages(
+            ControlFlowGraph cfg,
+            Compilation compilation,
+            ISymbol owningSymbol,
+            string typeToTrackMetadataName,
+            bool isNewInstanceFlagged,
+            string propertyToSetFlag,
+            bool isNullPropertyFlagged,
+            ImmutableHashSet<string> methodNamesToCheckForFlaggedUsage,
+            InterproceduralAnalysisKind interproceduralAnalysisKind = InterproceduralAnalysisKind.ContextSensitive,
+            bool pessimisticAnalysis = false)
+        {
+            var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilation);
+            var pointsToAnalysisResult = PointsToAnalysis.GetOrComputeResult(
+                cfg, owningSymbol, wellKnownTypeProvider, interproceduralAnalysisKind, pessimisticAnalysis);
+
+            var analysisContext = PropertySetAnalysisContext.Create(PropertySetAbstractValueDomain.Default,
+                wellKnownTypeProvider, cfg, owningSymbol, interproceduralAnalysisKind, pessimisticAnalysis, pointsToAnalysisResult, GetOrComputeResultForAnalysisContext,
+                typeToTrackMetadataName,
+                isNewInstanceFlagged,
+                propertyToSetFlag,
+                isNullPropertyFlagged,
+                methodNamesToCheckForFlaggedUsage);
+            var result = GetOrComputeResultForAnalysisContext(analysisContext);
+            return result.HazardousUsages;
+        }
+
+        private static PropertySetAnalysisResult GetOrComputeResultForAnalysisContext(PropertySetAnalysisContext analysisContext)
+        {
+            var operationVisitor = new PropertySetDataFlowOperationVisitor(analysisContext);
+            var analysis = new PropertySetAnalysis(BinaryFormatterAnalysisDomainInstance, operationVisitor);
+            return analysis.GetOrComputeResultCore(analysisContext, cacheResult: true);
```

Comment:
>cacheResult: true [](start = 68, length = 17)

We typically cache the result if there is another dataflow analysis sitting on top of an analysis (such as PointsTo, Copy, etc.). Otherwise, caching does not really help as you are unlikely to invoke the same analysis within the same compilation. Do you see multiple analysis invocations with same context for this analysis?

Reply:
I saw that, I'm guessing due to the different CFGs mentioned above.